### PR TITLE
sUSDS

### PR DESCRIPTION
**Pull request by jrhea**
This PR adds the tests for susds.

### CODE REVIEW

**Review by jalextowle — APPROVED** [review]
LGTM. One other thing would be adding the deployment script, which would be good to do in this PR.

File: tasks/deploy/config/mainnet/susds-182day.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                MAINNET_ERC4626_COORDINATOR_NAME,
+            ).address,
+            CONTRIBUTION,
+        ]);
+        let pc = await hre.viem.getPublicClient();
+        await pc.waitForTransactionReceipt({ hash: tx });
+    },
+    poolDeployConfig: async (hre) => {
+        return {
+            baseToken: DAI_ADDRESS_MAINNET,
```
Comment (jalextowle) [inline tasks/deploy/config/mainnet/susds-182day.ts:58]: Should this and the below be the USDS addresses?

File: tasks/deploy/config/mainnet/susds-182day.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    options: async (hre) => ({
+        extraData: "0x",
+        asBase: true,
+        destination: (await hre.getNamedAccounts())["deployer"] as Address,
+    }),
+    // Prepare to deploy the contract by setting approvals.
+    prepare: async (hre) => {
+        let baseToken = await hre.viem.getContractAt(
+            "contracts/src/interfaces/IERC20.sol:IERC20",
+            DAI_ADDRESS_MAINNET,
```
Comment (jalextowle) [inline tasks/deploy/config/mainnet/susds-182day.ts:45]: Should this be USDS?

File: tasks/deploy/config/mainnet/susds-182day.ts (hunk 1 above)
Comment (jrhea) [reply tasks/deploy/config/mainnet/susds-182day.ts:58]: yes

File: tasks/deploy/config/mainnet/susds-182day.ts (hunk 2 above)
Comment (jrhea) [reply tasks/deploy/config/mainnet/susds-182day.ts:45]: yes

**Pull request merged**